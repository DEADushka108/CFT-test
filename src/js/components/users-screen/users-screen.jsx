import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header';
import Footer from '../footer/footer';
import {getUsersList} from '../../store/users/selectors';
import {connect} from 'react-redux';
import {userDetails} from '../../types/user';
import UserCardSmall from '../user-card-small/user-card-small';

const UsersScreen = (props) => {
  const {usersList} = props;

  return <Fragment>
    <Header/>
    <main className="page-main">
      <h1 className="visually-hidden">users</h1>
      <section className="users-block">
        {usersList.map((userInfo) => {
          const {id} = userInfo;
          return <UserCardSmall key={id} userInfo={userInfo}/>;
        })}
      </section>
    </main>

    <Footer/>

  </Fragment>;
};

UsersScreen.propTypes = {
  usersList: PropTypes.arrayOf(userDetails),
};

const mapStateToProps = (state) => ({
  usersList: getUsersList(state),
});

export {UsersScreen};
export default connect(mapStateToProps)(UsersScreen);
