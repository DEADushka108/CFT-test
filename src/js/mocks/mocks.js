export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
