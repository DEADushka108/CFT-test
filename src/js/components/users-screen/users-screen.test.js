import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {UsersScreen} from './users-screen';
import {store, usersList} from '../../mocks/mocks.js';

it(`Should render UsersScreen component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <UsersScreen
            usersList={usersList}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    });

  expect(tree).toMatchSnapshot();
});
