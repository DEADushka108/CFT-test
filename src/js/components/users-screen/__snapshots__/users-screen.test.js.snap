// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UsersScreen component correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Chirrup
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Main
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/users"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Users
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contacts"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#message"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact us
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <h1
      className="visually-hidden"
    >
      users
    </h1>
    <section
      className="users-block"
    >
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Bret"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Leanne Graham
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/1"
                onClick={[Function]}
              >
                Bret
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Sincere@april.biz"
              >
                Sincere@april.biz
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:1-770-736-8031 x56442"
              >
                1-770-736-8031 x56442
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="hildegard.org"
              >
                hildegard.org
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Antonette"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Ervin Howell
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/2"
                onClick={[Function]}
              >
                Antonette
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Shanna@melissa.tv"
              >
                Shanna@melissa.tv
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:010-692-6593 x09125"
              >
                010-692-6593 x09125
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="anastasia.net"
              >
                anastasia.net
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Samantha"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Clementine Bauch
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/3"
                onClick={[Function]}
              >
                Samantha
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Nathan@yesenia.net"
              >
                Nathan@yesenia.net
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:1-463-123-4447"
              >
                1-463-123-4447
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="ramiro.info"
              >
                ramiro.info
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Karianne"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Patricia Lebsack
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/4"
                onClick={[Function]}
              >
                Karianne
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Julianne.OConner@kory.org"
              >
                Julianne.OConner@kory.org
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:493-170-9623 x156"
              >
                493-170-9623 x156
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="kale.biz"
              >
                kale.biz
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Kamren"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Chelsey Dietrich
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/5"
                onClick={[Function]}
              >
                Kamren
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Lucio_Hettinger@annie.ca"
              >
                Lucio_Hettinger@annie.ca
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:(254)954-1289"
              >
                (254)954-1289
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="demarco.info"
              >
                demarco.info
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Leopoldo_Corkery"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Mrs. Dennis Schulist
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/6"
                onClick={[Function]}
              >
                Leopoldo_Corkery
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Karley_Dach@jasper.info"
              >
                Karley_Dach@jasper.info
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:1-477-935-8478 x6430"
              >
                1-477-935-8478 x6430
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="ola.org"
              >
                ola.org
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Elwyn.Skiles"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Kurtis Weissnat
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/7"
                onClick={[Function]}
              >
                Elwyn.Skiles
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Telly.Hoeger@billy.biz"
              >
                Telly.Hoeger@billy.biz
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:210.067.6132"
              >
                210.067.6132
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="elvis.io"
              >
                elvis.io
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Maxime_Nienow"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Nicholas Runolfsdottir V
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/8"
                onClick={[Function]}
              >
                Maxime_Nienow
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Sherwood@rosamond.me"
              >
                Sherwood@rosamond.me
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:586.493.6943 x140"
              >
                586.493.6943 x140
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="jacynthe.com"
              >
                jacynthe.com
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Delphine"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Glenna Reichert
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/9"
                onClick={[Function]}
              >
                Delphine
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Chaim_McDermott@dana.io"
              >
                Chaim_McDermott@dana.io
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:(775)976-6794 x41206"
              >
                (775)976-6794 x41206
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="conrad.com"
              >
                conrad.com
              </a>
            </li>
          </ul>
        </div>
      </article>
      <article
        className="users-block__user-card-small user-card-small"
      >
        <div
          className="user-card-small__image"
        >
          <img
            alt="Moriah.Stanton"
            className="user-card-small__user-image"
            src="./img/content/no-user.png"
          />
          <div>
            <p
              className="user-card-small__name"
            >
              Clementina DuBuque
            </p>
          </div>
        </div>
        <div
          className="user-card-small__user-info"
        >
          <ul
            className="user-card-small__info-list"
          >
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Username:
              </p>
              <a
                className="user-card-small__link"
                href="/user/10"
                onClick={[Function]}
              >
                Moriah.Stanton
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Email:
              </p>
              <a
                className="user-card-small__link"
                href="mailto:Rey.Padberg@karina.biz"
              >
                Rey.Padberg@karina.biz
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Phone:
              </p>
              <a
                className="user-card-small__link"
                href="tel:024-648-3804"
              >
                024-648-3804
              </a>
            </li>
            <li
              className="user-card-small__item"
            >
              <p
                className="user-card-small__text"
              >
                Website:
              </p>
              <a
                className="user-card-small__link"
                href="ambrose.net"
              >
                ambrose.net
              </a>
            </li>
          </ul>
        </div>
      </article>
    </section>
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Chirrup
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          © 2020 Chirrup Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
